app/services/katello/pulp3/repository.rb

```diff
@@ -193,10 +193,12 @@ def delete(href = repository_reference.try(:repository_href))
         ignore_404_exception { api.repositories_api.delete(href) } if href
       end
 
-      def sync
+      def sync(options = {})
         sync_url_params = {remote: repo.remote_href, mirror: repo.root.mirror_on_sync}
-        skip_type_param = skip_types
-        sync_url_params[:skip_types] = skip_type_param if skip_type_param
+
+        sync_url_params[:optimize] = options[:optimize] if options.key?(:optimize) == false
```

Comment:
even thought optimize should always be false for non-yum repos, i think it makes sense  to move this to yum.rb